server/controllers/DocumentController.js

```diff
@@ -0,0 +1,183 @@
+        viewAccess: request.body.viewAccess,
+        ownerId: request.body.ownerId
+      })
+      .then(document => response.status(201).send({
+        document,
+        message: 'Document created successfully!'
+      }))
+      .catch(error => response.status(500).send({
+        error,
+        message: 'Something went wrong! The document could not be saved.'
+      }));
+  },
+  update(request, response) {
+    if (request.body.title === '' || request.body.content === ''
+      || request.body.viewAccess === '') {
+      return response.status(400).send({
+        message: 'Fields cannot be empty'
+      });
+    }
+    return Documents
+      .findById(parseInt(request.params.id, 10), {})
+      .then((document) => {
+        if (!document) {
+          return response.status(404).send({
+            message: 'Document Not Found'
+          });
+        }
+        return document
+          .update({
+            title: request.body.title,
+            content: request.body.content,
+            viewAccess: request.body.viewAccess
+          })
+          .then(() => response.status(201).send({
+            document,
+            message: 'Document successfully updated!'
+          }))
+          .catch(error => response.status(500).send({
+            error,
+            message: 'Something went wrong! The document could not be saved.'
+          }));
+      })
+      .catch(error => response.status(500).send({
+        error,
+        message: 'The document could not be saved.'
+      }));
+  },
+  delete(request, response) {
+    return Documents
+      .findById(parseInt(request.params.id, 10), {})
+      .then((document) => {
+        if (!document) {
+          return response.status(400).send({
+            message: 'Document not found'
+          });
+        }
+        return document
+          .destroy()
+          .then(() =>
+            response.status(200).send({
+              message: 'Document deleted successfully.'
+            })
+          )
+          .catch(error => response.status(500).send({
+            error,
+            message: 'Something went wrong! The document could not be saved.'
+          }));
+      })
+      .catch(error => response.status(500).send({
+        error,
+        message: 'The document could not be deleted.'
+      }));
+  },
+  getAll(request, response) {
+    Documents
+      .findAll()
+      .then(documents => response.status(200).send(documents))
+      .catch(error => response.status(400).send(error));
+  },
+  getMyDocuments(req, res) {
+    return Documents
+    .findAll({
+      where: {
+        $or: [
+          { viewAccess: 'public' },
+          {
+            role: String(req.decoded.data.roleId)
+          },
+          {
+            userId: req.params.id
+          }
+        ]
+      },
+      include: [Users],
+      order: [['updatedAt', 'DESC']]
+    })
+    .then((document) => {
+      if (!document) {
+        return res.status(404).send({
```

Comment:
There really can't be a 404 here since the user is trying to get all of his/her documents. You could just send back an empty array.